### PR TITLE
Muon analysis crash when using default grouping with no runs

### LINKED ISSUE #29705: instrument change bug
Open muon analysis
Set instrument to EMU
Load some data
Change the instrument to MUSR
Go to grouping tab
Click default - crash

### PR DESCRIPTION
**Pull request by AnthonyLim23**
**Description of work.**

**To test:**
1. Go to the grouping tab
2. Press default 
3. It used to crash,  you now get a pop up

Test 2:
Open muon analysis
Set instrument to EMU
Load some data
Change the instrument to MUSR
Go to grouping tab
Click default - crash

Fixes #29705. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/test/Muon/grouping_tab/grouping_tab_presenter_test.py [hunk 1]
```diff
@@ -238,6 +238,16 @@ def _run_handle_load_grouping_with_mocked_load(self, groups, pairs, description=
             mock_load.return_value = (groups, pairs, 'description', default)
             self.presenter.handle_load_grouping_from_file()
 
+    def test_default_clickd(self):
```
Comment (Matt-Cumber) [inline scripts/test/Muon/grouping_tab/grouping_tab_presenter_test.py:241]:
```suggestion
    def test_default_clicked(self):
```

**Review by Matt-Cumber — CHANGES_REQUESTED** [review]
Code looks good and it fixes the bug. Just one very minor comment

**Comment by martyngigg** [comment]
Is it worth us trying to get this into the build tonight for testing?

**Comment by AnthonyLim23** [comment]
I have already made a package and sent it to them

**Comment by martyngigg** [comment]
Cool. Thanks.

**Review by Matt-Cumber — APPROVED** [review]

**Pull request merged**